File: app/src/main/java/com/visionx/daggerpractice/di/CarComponent.java
package com.visionx.daggerpractice.di;

import com.visionx.daggerpractice.car.Car;
import com.visionx.daggerpractice.MainActivity;

import dagger.Component;

@Component (modules =  {WheelModeule.class , PertrolEngineModule.class})
interface CarComponent {
 Car getCar();
 void injects(MainActivity mainActivity);
}

File: app/src/main/java/com/visionx/daggerpractice/di/CarComponent.kt
package com.visionx.daggerpractice.di

import com.visionx.daggerpractice.MainActivity
import com.visionx.daggerpractice.car.Car
import dagger.Component

@Component(modules = [WheelModeule::class , DieselEngineModule::class])
 interface CarComponent {
    var car: Car?
    fun injects(mainActivity: MainActivity?)
}